server/interpreter.js

```diff
@@ -2085,7 +2097,7 @@ Interpreter.prototype['stepAssignmentExpression'] = function() {
   var node = state.node;
   if (!state.doneLeft_) {
     state.doneLeft_ = true;
-    stack.push({node: node['left'], components: true});
+    this.pushNode_(node['left']).components = true;
```

Comment:
`components`, `scope`, `throwValue`, and who knows what in the future.  I'm not sure that `this.pushNode_(node['foo'], null, null, value);` is more readable than `this.pushNode_(node['foo']).throwValue = value;`.